app/javascript/packs/document-capture-welcome.ts

```diff
@@ -0,0 +1,79 @@
+import { trackEvent } from '@18f/identity-analytics';
+import { hasCamera, isCameraCapableMobile } from '@18f/identity-device';
+
+const GRACE_TIME_FOR_CAMERA_CHECK_MS = 2000;
+const DEVICE_CHECK_EVENT = 'Idv: Mobile device and camera check';
+
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function measure<T>(
+  measureName: string,
+  promise: Promise<T>,
+): Promise<{ result: T; duration: number }> {
+  performance.measure(measureName);
+  const result = await promise;
+  const { duration } = performance.measure(measureName, {
+    end: performance.now(),
+  });
+  return { result, duration };
+}
+
+function addFormInputsForMobileDeviceCapabilities() {
+  const form = document.querySelector<HTMLFormElement>('.js-consent-continue-form');
+
+  if (!form) {
+    return;
+  }
+
+  if (!isCameraCapableMobile()) {
+    trackEvent(DEVICE_CHECK_EVENT, {
+      is_camera_capable_mobile: false,
+      user_agent: navigator.userAgent,
+    });
+    return;
+  }
+
+  // The check for a camera on the device is async -- kick it off here and intercept
+  // submit() to ensure that it completes in time.
+  const cameraCheckPromise = measure(DEVICE_CHECK_EVENT, hasCamera()).then(
+    ({ result: cameraPresent, duration }) => {
+      trackEvent(DEVICE_CHECK_EVENT, {
```

Comment:
You may want to `await` this since it's a network request that's not guaranteed to finish before the user leaves the page through the submission.

Reply:
Yeah, good call. I moved it after the dom modification just in case (it is more important to move the user on in the process than capture analytics here)